src/server/page_server.cpp

```diff
@@ -205,6 +206,13 @@ page_server::connection_handler::receive_start_catch_up (tran_server_conn_t::seq
 
   er_log_debug (ARG_FILE_LINE, "receive_start_catch_up: hostname = %s, port = %d, LSA = (%lld|%d)\n", host.c_str (), port,
 		LSA_AS_ARGS (&catchup_lsa));
+  if (port == -1)
```

Comment:
I don't think I understand you. The cachup process will be like this:
1. A new PS, follower, attaches its ATS
2. The ATS elects a leader PS, or it's been elected. The leader PS is reliable enough to catch up.
3. The ATS sends this information to the follower PS.
4. The follower PS requests the catch-up the leader PS based on the information. It happens in this function, receive_start_catch_up(). The catch-up process will be taken by another thread which will be initiated here to avoid blocking this PS's request handler for the ATS.
6. A connection between two PSs will be established.

Could you point out where you feels weird?

Reply:
I think, what I meant is the distinction between "cold start" and "catchup start" of PS.
- at cold start, the ATS can decide - in `active_tran_server::connection_handler::on_connecting ` - to skip the catchup step - of course PS connector has to support this
- at "hot/catchup start" the ATS will dispatch the catchup message in `active_tran_server::connection_handler::on_connecting`

But, please disregard this message as I don't want to confuse you even further. :)